src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -485,11 +485,17 @@ bool Unit::IsWithinMeleeRange(Unit const* obj) const
     float dz = GetPositionZMinusOffset() - obj->GetPositionZMinusOffset();
     float distsq = dx*dx + dy*dy + dz*dz;
 
-    float maxdist = GetCombatReach() + obj->GetCombatReach() + 4.0f / 3.0f;
+    float maxdist = GetMeleeRange(obj);
 
     return distsq <= maxdist * maxdist;
 }
 
+float Unit::GetMeleeRange(Unit const* target) const
+{
+    float range = this->GetCombatReach() + target->GetCombatReach() + 4.0f / 3.0f;
```

Comment:
Because it's not needed and causes confusion when mixed with code that doesn't use it (as in the rest of TC sources).